finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/service/H2ReqRepFrame.scala

```diff
@@ -32,7 +33,7 @@ case class H2ReqRep(request: Request, response: Try[RepAndFrame]) {
   override def toString: String = s"H2ReqRep($request, $responseOnly, finalFrame=$finalFrame)"
 }
 
-object H2ReqRep {
+object H2ReqRepFrame {
   // type aliases to minimise typing
   type RepAndFrame = (Response, FinalFrame)
   type FinalFrame = Option[Try[Frame]]
```

Comment:
I think so.  The responseClassifier only looks at the request and response.  The stream classifier looks at the final frame as well, which may or may not exist.  Technically, once the response is complete we'll be able to tell if the stream is empty and could fold the empty stream case into early classification but... conceptually it feels cleaner to handle the empty stream classification as part of stream classification.  By doing this, responseClassifiers never have to look at response.stream at all.